### PR TITLE
Allow partner application private key to be passed as a string

### PR DESCRIPTION
**Pull request by stephenminded**
Allow partner application private key to be passed as a string or a path to a file.

For applications that are hosted on platforms like heroku, it's handy to
be able to manage the application private key in an environment
variable. This change allows the client of PartnerApplication to
instantiate it with either a string containing the private key, or a
path to a file (existing behaviour).

@Shopify/platform

### CODE REVIEW

File: lib/xeroizer/partner_application.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def initialize(consumer_key, consumer_secret, path_to_private_key, ssl_client_ce
           :signature_method => 'RSA-SHA1'
         }
         options = default_options.merge(options).merge(
-          :private_key_file => path_to_private_key,
+          :consumer_secret => read_certificate(path_to_private_key),
```
Comment (christhomson) [inline lib/xeroizer/partner_application.rb]: Where does `path_to_private_key` come from now that the arg has been renamed?

Comment (stephenminded) [inline lib/xeroizer/partner_application.rb]: It doesn't come from anywhere because that's a screw up on my behalf :dizzy_face:

**Comment by kwestein** [comment]
Should we PR this on the actual xeroizer repo?

File: lib/xeroizer/partner_application.rb (hunk 1 above)
Comment (shawnfrench) [inline lib/xeroizer/partner_application.rb]: Can't you just (ab)use xeroizer as is by passing nil for `path_to_private_key ` and in options add `consumer_secret`? Did you try?

Comment (stephenminded) [inline lib/xeroizer/partner_application.rb]: > Can't you just (ab)use xeroizer as is by passing nil for path_to_private_key and in options add consumer_secret? 

Yeah, I could do that.

**Comment by stephenminded** [comment]
Closing this, I just went with Shawn's suggestion.

**Pull request closed without merging**